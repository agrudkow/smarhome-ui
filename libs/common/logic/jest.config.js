module.exports = {
  name: 'common-logic',
  preset: '../../../jest.config.js',
  transform: {
    '^.+\\.[tj]sx?$': 'ts-jest',
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx', 'html'],
  coverageDirectory: '../../../coverage/libs/common/logic',
};
