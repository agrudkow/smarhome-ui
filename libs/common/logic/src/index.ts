export * from './lib/use-accordion';
export * from './lib/use-window-dimensions';
