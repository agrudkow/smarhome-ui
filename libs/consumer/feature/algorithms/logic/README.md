# consumer-feature-algorithms-logic

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test consumer-feature-algorithms-logic` to execute the unit tests via [Jest](https://jestjs.io).
