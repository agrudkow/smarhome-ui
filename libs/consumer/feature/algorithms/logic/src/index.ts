export * from './lib/algorithm.interface';
export * from './lib/algorithms-cells-parser';
export * from './lib/algorithms-data-parser';
export * from './lib/use-algorithms-list';
export * from './lib/use-algorithm-detials';
export * from './lib/use-algorithm-rating';
export * from './lib/use-select-algorithm';
