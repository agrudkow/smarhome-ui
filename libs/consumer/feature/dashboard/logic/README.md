# consumer-feature-dashboard-logic

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test consumer-feature-dashboard-logic` to execute the unit tests via [Jest](https://jestjs.io).
