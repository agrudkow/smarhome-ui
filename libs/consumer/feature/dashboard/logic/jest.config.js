module.exports = {
  name: 'consumer-feature-dashboard-logic',
  preset: '../../../../../jest.config.js',
  transform: {
    '^.+\\.[tj]sx?$': 'ts-jest',
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx', 'html'],
  coverageDirectory:
    '../../../../../coverage/libs/consumer/feature/dashboard/logic',
};
