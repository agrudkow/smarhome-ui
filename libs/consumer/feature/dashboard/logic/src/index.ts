export * from './lib/use-billings';
