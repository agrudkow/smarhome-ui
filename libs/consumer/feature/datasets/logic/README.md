# consumer-feature-datasets-logic

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test consumer-feature-datasets-logic` to execute the unit tests via [Jest](https://jestjs.io).
