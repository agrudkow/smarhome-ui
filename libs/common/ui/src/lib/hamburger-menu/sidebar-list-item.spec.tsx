import React from 'react';
import {
  BrowserRouter as Router,
  Router as RouterWithHistory,
} from 'react-router-dom';
import { createMemoryHistory } from 'history';
import SidebarListItem from './sidebar-list-item';
import { customRender } from '@smarthome/common/logic';

describe(' SidebarListItem', () => {
  it('should render successfully', () => {
    const { baseElement } = customRender(
      <Router>
        <SidebarListItem
          text=""
          showText={true}
          icon={<div />}
          linkRoute="/test"
        />
      </Router>
    );
    expect(baseElement).toBeTruthy();
  });

  it('should match snapshot when expanded', () => {
    const { baseElement } = customRender(
      <Router>
        <SidebarListItem
          text=""
          showText={true}
          icon={<div />}
          linkRoute="/test"
        />
      </Router>
    );
    expect(baseElement.firstChild).toMatchSnapshot();
  });

  it('should not show text when "showTest" set to false', () => {
    const { getByTestId } = customRender(
      <Router>
        <SidebarListItem
          text=""
          showText={false}
          icon={<div />}
          linkRoute="/test"
        />
      </Router>
    );

    expect(getByTestId('text')).toHaveStyleRule('max-width', '0');
    expect(getByTestId('text')).toHaveStyleRule('opacity', '0');
    expect(getByTestId('text')).toHaveStyleRule('padding-left', '0');
  });

  it('should highlight button component when link route matches with history path', () => {
    const testPath = '/test-path';
    const history = createMemoryHistory();
    history.push(testPath);
    const { getByTestId } = customRender(
      <RouterWithHistory history={history}>
        <SidebarListItem
          text=""
          showText={false}
          icon={<div />}
          linkRoute={testPath}
        />
      </RouterWithHistory>
    );

    expect(getByTestId('styled-sidebar-list-item')).not.toHaveStyleRule(
      'background-color',
      'transparent'
    );
    expect(getByTestId('styled-sidebar-list-item')).not.toHaveStyleRule(
      'border-left',
      '2px solid transparent'
    );
  });
});
