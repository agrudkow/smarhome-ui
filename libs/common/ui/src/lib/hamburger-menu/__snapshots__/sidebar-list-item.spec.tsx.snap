// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` SidebarListItem should match snapshot when expanded 1`] = `
.c0 {
  height: 45px;
  padding: 12px 16px;
  background-color: transparent;
  border-left: 2px solid transparent;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transition: background-color 0.3s ease-in-out,border-left 0.3s ease-in-out;
  transition: background-color 0.3s ease-in-out,border-left 0.3s ease-in-out;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:hover {
  background-color: #4E4E4E;
}

.c2 {
  color: #555;
  font-size: 14px;
  line-height: 20px;
  -webkit-letter-spacing: -0.3px;
  -moz-letter-spacing: -0.3px;
  -ms-letter-spacing: -0.3px;
  letter-spacing: -0.3px;
  color: #DFE3D5;
  overflow: hidden;
  white-space: nowrap;
  padding-top: 2px;
  -webkit-transition: max-width 0.3s ease-in-out,opacity 0.3s ease-in-out, padding-left 0.3s ease-in-out,color 0.3s ease-in-out;
  transition: max-width 0.3s ease-in-out,opacity 0.3s ease-in-out, padding-left 0.3s ease-in-out,color 0.3s ease-in-out;
  max-width: 220px;
  opacity: 1;
  padding-left: 20px;
}

.c1 {
  min-width: 24px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 path {
  -webkit-transition: fill 0.3s ease-in-out;
  transition: fill 0.3s ease-in-out;
  fill: #DFE3D5;
}

@media (min-width:769px) {
  .c2 {
    font-size: 14px;
    line-height: 20px;
  }
}

@media (min-width:1024px) {
  .c2 {
    font-size: 16px;
    line-height: 24px;
  }
}

@media (min-width:1024px) {
  .c2 {
    font-size: 20px;
    line-height: 26px;
  }
}

<div>
  <li>
    <a
      class="c0"
      data-testid="styled-sidebar-list-item"
      href="/test"
    >
      <div
        class="c1"
      >
        <div />
      </div>
      <div
        class="c2"
        data-testid="text"
      />
    </a>
  </li>
</div>
`;
